### PR TITLE
Add filtering residents by student group

### LINKED ISSUE #93: Filter residents by student group

### PR DESCRIPTION
**Pull request by teekoksiang**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -49,6 +56,9 @@ public FindCommand parse(String args) throws ParseException {
             predicates.add(getBlockPredicate(argMultimap.getValue(PREFIX_BLOCK).get()));
         }
 
+        parseStudentGroupsForFind(argMultimap.getAllValues(PREFIX_STUDENT_GROUP))
+            .ifPresent(studentGroupSet -> predicates.add(new StudentGroupPredicate(studentGroupSet)));
+
```
Comment (yanchenglee98) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java]: Is there a reason you did not follow the previous standard of 
`if (argMultimap.getValue(PREFIX_STUDENT_GROUP).isPresent()) {
      predicates.add(getStudentGroupPredicate(argMultimap.getValue(PREFIX_STUDENT_GROUP).get()));
 }`?

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 2]
```diff
@@ -73,4 +83,22 @@ public FindCommand parse(String args) throws ParseException {
         assert nameKeywords.length > 0 : "there should be some keywords";
         return new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords));
     }
+
+    /**
+     * Parses {@code Collection<String> studentGroups} into a {@code Set<StudentGroup>} if {@code studentGroups}
+     * is non-empty.
+     * If {@code studentGroups} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<StudentGroup>} containing zero studentGroups.
```
Comment (yanchenglee98) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:95]: Missing a JavaDoc comment that explains when a `ParseException` is thrown

**Review by yanchenglee98 — COMMENTED** [review]
Looks good. Just some minor nits on standardization.

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java (hunk 1 above)
Comment (teekoksiang) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java]: Not really, updated the pr to follow the convention. Thanks.

**Review by yanchenglee98 — APPROVED** [review]
Good work. LGTM

**Pull request merged**